File: binary_search.py
def binary_search(arr,start,end,item):
    if end>=start:
        mid = (start+end)//2

        if arr[mid] == item:
            return mid

        elif arr[mid] > item:
            return binary_search(arr,start,mid-1,item)
        
        elif arr[mid] < item:
            return binary_search(arr,mid+1,end,item)
        else:
            return -1
        
arr = [2,4,6,8,10,12,14,16]
item = 6
result = binary_search(arr,0,len(arr)-1,item)

if result != -1:
    print("Elements found on index: ",str(result))
else:
    print("Elements not found")

File: C Program/Searching Technique/Binary_Search.c
#include<stdio.h>

int binarysearch(int arr[],int key,int low,int high){
    while(low<=high){
        int mid = (low+high)/2;
        if(arr[mid] == key){
            return key;
        }
        if(arr[mid] < key){
            low = mid+1;
        }
        else{
            high = mid - 1;
        }
    }
    return -1;
}

int main(){
    int arr[] = {1,2,3,4,5,6,7,8,9,10};
    int n = sizeof(arr)/sizeof(arr[0]);
    int key;
    printf("Enter a key to be searched:");
    scanf("%d",&key);
    if(binarysearch(arr,key,0,n-1) == -1){
        printf("Not Found");
    }
    else{
        printf("Found");
    }
    return 0;
}